PetProject/Screens/SearchStackScreens/SearchScreen/ViewModel+VC/SearchScreenViewModel.swift

```diff
@@ -0,0 +1,153 @@
+                self?.searchHistory = savedReqests
+            }
+            .store(in: &cancellables)
+        self.fetchStores()
+    }
+    
+    func searchDeals() {
+        searchDispathQueue.async { [weak self] in
+            guard let self = self else { return }
+            self.searchDispathGroup.wait()
+            self.searchDispathGroup.enter()
+            self.apiService.globalSearchDeals(
+                title: self.searchText,
+                onPaginationPage: self.paginationPage
+            )?
+                .receive(on: DispatchQueue.main)
+                .sink(
+                    receiveCompletion: { completion in
+                        switch completion {
+                        case .failure(let error):
+                            self.searchDispathGroup.leave()
+                            switch error {
+                            case .decodeError(let description):
+                                print("Global search deals JSON decode error: \(String(describing: description))")
+                            default:
+                                print("Global search deals error: \(error)")
+                            }
+                        case .finished:
+                            print("Global search deals finished")
+                        }
+                    },
+                    receiveValue: { deals in
+                        self.searchedDeals += deals.filter {
+                            !self.searchedDeals.contains($0)
+                        }
+                        self.paginationIsEnded = deals.isEmpty
+                        self.searchDispathGroup.leave()
+                        self.paginationPage += 1
+                        deals.forEach({ self.fetchDealImage($0) })
+                        if !self.searchHistory.contains(where: { $0.searchText == self.searchText }) {
+                            self.saveHistoryService.saveSearchReqest(SearchReqestModel(searchText: self.searchText))
+                        }
+                    }
+                )
+                .store(in: &self.cancellables)
+        }
+    }
+    
+    func fetchDealImage(_ deal: DealModel) {
+        apiService.fetchImage(forDeal: deal)?
+            .receive(on: DispatchQueue.main)
+            .sink(
+                receiveCompletion: { completion in
+                    switch completion {
+                    case .failure:
+                        print("Faild to fetch deal image")
+                    case .finished:
+                        print("Fetch deal image finished")
+                    }
+                },
+                receiveValue: { [weak self] image in
+                    self?.dealsLogos[deal.id] = image
+                }
+            )
+            .store(in: &cancellables)
+    }
+
+    func deleteSearchResult() {
+        searchText = ""
+        searchedDeals = []
+        paginationPage = 0
+        paginationIsEnded = false
```

Comment:
Just marafet stuff
when you using bool var in context when something happened or not, try to place `is` word in the first place, it will be little easier to read
`isSomethingHappend`
`isMyComputerExploded`, etc.

Reply:
Done